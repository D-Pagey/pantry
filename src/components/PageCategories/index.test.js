import React from 'react';
import { PageCategories } from '.';
import { CategoriesArray, Fridge } from '../../fixtures';

const props = {};

const context = {
  isAuthed: false,
  isCheckingAuth: true,
  categories: CategoriesArray,
  fridge: Fridge
};

describe('PageCategories component', () => {
  it('should render', () => {
    const { container } = render(<PageCategories {...props} />, context);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render loading spinner if checking auth', () => {
    const { queryByTestId } = render(<PageCategories {...props} />, context);
    expect(queryByTestId('pageCategories')).toBeNull();
  });

  it('should render category list if authed', () => {
    const { getByTestId } = render(<PageCategories {...props} />, {
      ...context,
      isAuthed: true,
      isCheckingAuth: false,
    });
    getByTestId('categoryList');
  });
});
