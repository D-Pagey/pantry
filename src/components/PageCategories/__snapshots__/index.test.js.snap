// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageCategories component should render 1`] = `
.c0 {
  padding: 50px 0;
}

<div
  class="c0"
>
  <div
    data-testid="loading"
  >
    <div
      class="css-9ebb9a"
    >
      <div
        class="css-1m2ccf8"
      />
      <div
        class="css-x0lzl3"
      />
      <div
        class="css-3nmchs"
      />
      <div
        class="css-r7gb"
      />
      <div
        class="css-vr5200"
      />
      <div
        class="css-1p14snf"
      />
    </div>
  </div>
</div>
`;
